shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -21,4 +21,144 @@
+      CGPathAddLineToPoint(mutablePathRef, nil, clipSkRect.fRight - topRightRadii.x(),
+                           clipSkRect.fTop);
+      CGPathAddCurveToPoint(mutablePathRef, nil, clipSkRect.fRight, clipSkRect.fTop,
+                            clipSkRect.fRight, clipSkRect.fTop + topRightRadii.y(),
+                            clipSkRect.fRight, clipSkRect.fTop + topRightRadii.y());
+      // Move point vertically down to the bottom right corner and add the bottom right curve.
+      CGPathAddLineToPoint(mutablePathRef, nil, clipSkRect.fRight,
+                           clipSkRect.fBottom - bottomRightRadii.y());
+      CGPathAddCurveToPoint(mutablePathRef, nil, clipSkRect.fRight, clipSkRect.fBottom,
+                            clipSkRect.fRight - bottomRightRadii.x(), clipSkRect.fBottom,
+                            clipSkRect.fRight - bottomRightRadii.x(), clipSkRect.fBottom);
+      // Move point horizontally left to the bottom left corner and add the bottom left curve.
+      CGPathAddLineToPoint(mutablePathRef, nil, clipSkRect.fLeft + bottomLeftRadii.x(),
+                           clipSkRect.fBottom);
+      CGPathAddCurveToPoint(mutablePathRef, nil, clipSkRect.fLeft, clipSkRect.fBottom,
+                            clipSkRect.fLeft, clipSkRect.fBottom - bottomLeftRadii.y(),
+                            clipSkRect.fLeft, clipSkRect.fBottom - bottomLeftRadii.y());
+      // Move point vertically up to the top left corner and add the top left curve.
+      CGPathAddLineToPoint(mutablePathRef, nil, clipSkRect.fLeft,
+                           clipSkRect.fTop + topLeftRadii.y());
+      CGPathAddCurveToPoint(mutablePathRef, nil, clipSkRect.fLeft, clipSkRect.fTop,
+                            clipSkRect.fLeft + topLeftRadii.x(), clipSkRect.fTop,
+                            clipSkRect.fLeft + topLeftRadii.x(), clipSkRect.fTop);
+      CGPathCloseSubpath(mutablePathRef);
+
+      pathRef = mutablePathRef;
+      break;
+    }
+  }
+  // TODO(cyanglaz): iOS does not seem to support hard edge on CAShapeLayer. It clearly stated that
+  // the CAShaperLayer will be drawn antialiased. Need to figure out a way to do the hard edge
+  // clipping on iOS.
+  CAShapeLayer* clip = [[CAShapeLayer alloc] init];
+  clip.path = pathRef;
+  self.layer.mask = clip;
+  CGPathRelease(pathRef);
+}
+
+- (void)performClip:(flutter::MutatorType)type
+               rect:(const SkRect&)rect
+              rrect:(const SkRRect&)rrect
+               path:(const SkPath&)path {
+  FML_CHECK(type == flutter::clip_rect || type == flutter::clip_rrect ||
+            type == flutter::clip_path);
+  switch (type) {
+    case flutter::clip_rect:
+      [self clipRect:rect];
+      break;
+    case flutter::clip_rrect:
+      [self clipRRect:rrect];
+      break;
+    case flutter::clip_path:
+      // TODO(cyanglaz): Add clip path
+      break;
+    default:
+      break;
+  }
+}
+
+// Only acknowlege touches are inside if the touches are acknowlege inside by any of its
```

Comment:
Suggestion for alternative comment:

The ChildClippingView is as big as the FlutterView, we only want touches to be hit tested and consumed by this view if they are inside the smaller child view.